### PR TITLE
Allow `next` step to be a function

### PR DESCRIPTION
**Pull request by gavboulton**

### CODE REVIEW

File: lib/controller/mixins/next-step.js [hunk 1]
```diff
@@ -82,6 +82,10 @@ module.exports = Controller => class extends Controller {
             });
         }
 
+        if (typeof nextStep === 'function') {
+            nextStep = nextStep(req, res);
```
Comment (HughePaul) [inline lib/controller/mixins/next-step.js]: can you run it in the context of the controller, as that allows calling controller methods.

Comment (HughePaul) [inline lib/controller/mixins/next-step.js]: perhaps also change `if (typeof condition === 'string') {` to `if (typeof condition === 'string' || typeof condition === 'function') {` on line 66 to allow defaults in sets to also be functions and not having to do the fn true trick

Comment (HughePaul) [inline lib/controller/mixins/next-step.js]: or `if (typeof condition !== 'object') {`

File: lib/controller/mixins/next-step.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ module.exports = Controller => class extends Controller {
             let conditions = nextStep;
             nextStep = null;
             _.find(conditions, condition => {
-                if (typeof condition === 'string') {
+                if (typeof condition === 'string' || typeof condition === 'function') {
```
Comment (gavboulton) [inline lib/controller/mixins/next-step.js]: @HughePaul I've added a function check.

File: lib/controller/mixins/next-step.js [hunk 3]
```diff
@@ -82,6 +82,10 @@ module.exports = Controller => class extends Controller {
             });
         }
 
+        if (typeof nextStep === 'function') {
+            nextStep = nextStep.call(this, req, res);
```
Comment (gavboulton) [inline lib/controller/mixins/next-step.js]: @HughePaul Calling with controller context.

**Comment by gavboulton** [comment]
hmpo-form-wizard@8.1.0 published.

**Pull request merged**